pya/asig.py

```diff
@@ -133,6 +128,20 @@ def cn(self, val):
                         self.channels)
                 )
 
+    @property
+    def samples(self) -> int:
+        """
+        Return the length of signal in samples
+        """
+        return np.shape(self.sig)[0]
+
+    @property
+    def dur(self) -> float:
+        """
+        Return the duration in seconds
+        """
+        return self.samples / self.sr
+
```

Comment:
are `self.samples` and `self.sig` guaranteed to be not `None`? Otherwise this requires some safeguards